### PR TITLE
fix: [CDS-82305]: API to return all environments for the given list of projects within an org

### PR DESCRIPTION
**Pull request by sarthakkasat**
## Describe your changes
We are adding a support for the API which will take orgId and list of projectIds as query parameters and return all the environments which are child of given project

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54818)

### CODE REVIEW

File: 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java [hunk 1]
```diff
@@ -108,6 +108,71 @@ public Criteria createCriteriaForGetList(
...
+  public Criteria createCriteriaToGetScopedEnvironments(
+      String accountId, String orgIdentifier, List<String> projectIdentifiers, String envType, List<String> scopes) {
+    if (isEmpty(accountId)) {
+      throw new InvalidRequestException("account identifier cannot be null for environment list");
+    }
+    List<Criteria> orCriteriaList = new ArrayList<>();
+    Criteria accountCriteria;
+    Criteria orgCriteria;
+    Criteria projectCriteria;
+    if (isNotEmpty(scopes)) {
```
Comment (puthrayaharness) [inline 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java:120]: return early from such cases and you don't need else. Avoid nesting if else. Always return early if possible.
For example, I would write this as `if (isEmpty(scopes)) {throw Execption}; continue with logic without else`

File: 960-ng-core-beans/src/main/java/io/harness/ng/core/environment/dto/ProjectsEnvironmentResponseDTO.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import io.swagger.v3.oas.annotations.media.Schema;
+import java.util.Map;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Data;
+import lombok.experimental.FieldDefaults;
+
+@CodePulse(module = ProductModule.CDS, unitCoverageRequired = true, components = {HarnessModuleComponent.CDS_PIPELINE})
```
Comment (puthrayaharness) [inline 960-ng-core-beans/src/main/java/io/harness/ng/core/environment/dto/ProjectsEnvironmentResponseDTO.java:29]: coverage is false for DTO classes

Comment (puthrayaharness) [reply 960-ng-core-beans/src/main/java/io/harness/ng/core/environment/dto/ProjectsEnvironmentResponseDTO.java:29]: Compoent is Service and Environment

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

**Review by mohitg0795 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54818)* status: 5 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/environment/resources/EnvironmentResourceV2.java` line 507 at r1](https://reviewable.io/reviews/harness/harness-core/54818#-Ni4dQyoAWnto0vZGihB:-Ni4dQyp2i-WmXrctdv3:bw34k24) ([raw file](https://github.com/harness/harness-core/blob/d04aee2084b56050b84418f823aa81b1bb89e4f4/120-ng-manager/src/main/java/io/harness/ng/core/environment/resources/EnvironmentResourceV2.java#L507)):*
> ```Java
> 
>   @GET
>   @Path("listV3")
> ```

can we improve the path? Its not a good way to name api paths.

**Review by mohitg0795 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54818)* status: 5 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/environment/resources/EnvironmentResourceV2.java` line 511 at r1](https://reviewable.io/reviews/harness/harness-core/54818#-Ni4dc_V2Ii10TYxeazx:-Ni4dc_V2Ii10TYxeazy:b-8zq7ga) ([raw file](https://github.com/harness/harness-core/blob/d04aee2084b56050b84418f823aa81b1bb89e4f4/120-ng-manager/src/main/java/io/harness/ng/core/environment/resources/EnvironmentResourceV2.java#L511)):*
> ```Java
>   @InternalApi
>   @ApiOperation(value = "Get environments for given scopes", nickname = "getScopedEnvironments")
>   @Operation(operationId = "getScopedEnvironments", summary = "Get environments for given scopes",
> ```

lets avoid adding all these annotations as its already hidden and internal api. This will help make code more readable and reduce confusions.

**Review by mohitg0795 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54818)* status: 5 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java` line 135 at r1](https://reviewable.io/reviews/harness/harness-core/54818#-Ni4dzbQ6UGniVoU-6XF:-Ni4dzbQ6UGniVoU-6XG:b-gb7lwf) ([raw file](https://github.com/harness/harness-core/blob/d04aee2084b56050b84418f823aa81b1bb89e4f4/127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java#L135)):*
> ```Java
>         orCriteriaList.add(accountCriteria);
>       }
>       if (scopes.contains("Org")) {
> ```

does it need to be matched case sensitive? Or should it be case-insensitive?

File: 120-ng-manager/src/main/java/io/harness/ng/core/environment/resources/EnvironmentResourceV2.java [hunk 3]
```diff
@@ -502,6 +503,38 @@ private boolean checkFeatureFlagForOverridesV2(String accountId) {
     return getEnvironmentsPageByCriteria(criteria, pageRequest);
   }
 
+  @GET
+  @Path("listV3")
```
Comment (sribalij) [inline 120-ng-manager/src/main/java/io/harness/ng/core/environment/resources/EnvironmentResourceV2.java:507]: The path usually would be something like `/v3/list`. Can you please change this?

File: 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java (hunk 1 above)
Comment (sribalij) [reply 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java:120]: +1

File: 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java [hunk 4]
```diff
@@ -108,6 +108,71 @@ public Criteria createCriteriaForGetList(
...
+      if (scopes.contains("Account")) {
```
Comment (sribalij) [inline 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java:121]: Have static constants

**Review by sarthakkasat — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54818)* status: 5 file(s) to review. 1 change requests. 8. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/environment/resources/EnvironmentResourceV2.java` line 507 at r1](https://reviewable.io/reviews/harness/harness-core/54818#-Ni4dQyoAWnto0vZGihB:-Ni8cIXt60Aqsni6a0OZ:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/d04aee2084b56050b84418f823aa81b1bb89e4f4/120-ng-manager/src/main/java/io/harness/ng/core/environment/resources/EnvironmentResourceV2.java#L507)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

can we improve the path? Its not a good way to name api paths.
</blockquote></details>

Done.
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/environment/resources/EnvironmentResourceV2.java` line 507 at r1](https://reviewable.io/reviews/harness/harness-core/54818#gh-1378409529:-Ni8cIsj5I3VYCmZSSTe:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/d04aee2084b56050b84418f823aa81b1bb89e4f4/120-ng-manager/src/main/java/io/harness/ng/core/environment/resources/EnvironmentResourceV2.java#L507)):*
<details><summary><i>Previously, sribalij (Sridhar) wrote…</i></summary><blockquote>

The path usually would be something like `/v3/list`. Can you please change this?
</blockquote></details>

Done.
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/environment/resources/EnvironmentResourceV2.java` line 511 at r1](https://reviewable.io/reviews/harness/harness-core/54818#-Ni4dc_V2Ii10TYxeazx:-Ni8cKa80twYQzELoO02:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/d04aee2084b56050b84418f823aa81b1bb89e4f4/120-ng-manager/src/main/java/io/harness/ng/core/environment/resources/EnvironmentResourceV2.java#L511)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

lets avoid adding all these annotations as its already hidden and internal api. This will help make code more readable and reduce confusions.
</blockquote></details>

Done.
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java` line 120 at r1](https://reviewable.io/reviews/harness/harness-core/54818#gh-1377487005:-Ni8cdPZEdNMX2Iplens:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/d04aee2084b56050b84418f823aa81b1bb89e4f4/127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java#L120)):*
<details><summary><i>Previously, sribalij (Sridhar) wrote…</i></summary><blockquote>

+1
</blockquote></details>

Done.
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java` line 121 at r1](https://reviewable.io/reviews/harness/harness-core/54818#gh-1378410068:-Ni8eu3o-xXcb9TqOQYn:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/d04aee2084b56050b84418f823aa81b1bb89e4f4/127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java#L121)):*
<details><summary><i>Previously, sribalij (Sridhar) wrote…</i></summary><blockquote>

Have static constants
</blockquote></details>

Done.
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java` line 135 at r1](https://reviewable.io/reviews/harness/harness-core/54818#-Ni4dzbQ6UGniVoU-6XF:-Ni8gv0P2F1sks2eAirS:b-s2upja) ([raw file](https://github.com/harness/harness-core/blob/d04aee2084b56050b84418f823aa81b1bb89e4f4/127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java#L135)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

does it need to be matched case sensitive? Or should it be case-insensitive?
</blockquote></details>

We can make it case insensitive, but will that check be expensive?
___
*[`960-ng-core-beans/src/main/java/io/harness/ng/core/environment/dto/ProjectsEnvironmentResponseDTO.java` line 29 at r1](https://reviewable.io/reviews/harness/harness-core/54818#gh-1377487456:-Ni8fJZ_4bomliXuRzJw:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/d04aee2084b56050b84418f823aa81b1bb89e4f4/960-ng-core-beans/src/main/java/io/harness/ng/core/environment/dto/ProjectsEnvironmentResponseDTO.java#L29)):*
<details><summary><i>Previously, puthrayaharness (Deepak Puthraya) wrote…</i></summary><blockquote>

Compoent is Service and Environment
</blockquote></details>

Done.
___
*[`960-ng-core-beans/src/main/java/io/harness/ng/core/environment/dto/ProjectsEnvironmentResponseDTO.java` line 35 at r1](https://reviewable.io/reviews/harness/harness-core/54818#-Ni4e88T8x1WAIF6Gdd3:-Ni8fKQm2VyupBdISWa1:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/d04aee2084b56050b84418f823aa81b1bb89e4f4/960-ng-core-beans/src/main/java/io/harness/ng/core/environment/dto/ProjectsEnvironmentResponseDTO.java#L35)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

remove schema annotation as its an internal dto
</blockquote></details>

Done.

File: 960-ng-core-beans/src/main/java/io/harness/ng/core/environment/dto/ScopedEnvironmentResponseDTO.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2021 Harness Inc. All rights reserved.
```
Comment (lovish1999) [inline 960-ng-core-beans/src/main/java/io/harness/ng/core/environment/dto/ScopedEnvironmentResponseDTO.java:2]: :nit: 2023

File: 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java [hunk 6]
```diff
@@ -62,6 +62,9 @@
 @AllArgsConstructor(access = AccessLevel.PACKAGE, onConstructor = @__({ @Inject }))
 public class EnvironmentFilterHelper {
   @Inject private final FilterService filterService;
+  private static final String ACCOUNT = "ACCOUNT";
```
Comment (puthrayaharness) [inline 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java:65]: We probably have the scopes already defined as constants, let's leverage that. No need to recreate these constants.

File: 120-ng-manager/src/main/java/io/harness/ng/core/environment/resources/EnvironmentResourceV2.java [hunk 7]
```diff
@@ -502,6 +504,29 @@ private boolean checkFeatureFlagForOverridesV2(String accountId) {
...
+      @Parameter(description = NGCommonEntityConstants.PAGE_PARAM_MESSAGE) @QueryParam(
+          NGCommonEntityConstants.PAGE) @DefaultValue("0") int page,
+      @Parameter(description = NGCommonEntityConstants.SIZE_PARAM_MESSAGE) @QueryParam(
+          NGCommonEntityConstants.SIZE) @DefaultValue("100") int size,
+      @Parameter(description = NGCommonEntityConstants.ACCOUNT_PARAM_MESSAGE) @NotNull @QueryParam(
+          NGCommonEntityConstants.ACCOUNT_KEY) @AccountIdentifier String accountId,
+      @Parameter(description = "Environment Type") @QueryParam("envType") @ResourceIdentifier String envType,
+      @RequestBody(description = "This is the body for list of orgIds, projectIds and Scopes.")
+      ScopedEnvironmentRequestDTO scopedEnvironmentRequestDTO) {
+    Criteria criteria = environmentFilterHelper.createCriteriaToGetScopedEnvironments(accountId,
```
Comment (puthrayaharness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/environment/resources/EnvironmentResourceV2.java:521]: Move the logic to `environmentService`. Avoid constructing or stitching code in the resource layer.

Comment (puthrayaharness) [reply 120-ng-manager/src/main/java/io/harness/ng/core/environment/resources/EnvironmentResourceV2.java:521]: I mean move the fetching criteria from environmentFilter service and then create the pageRequest and then call environmentService.

Instead, just call environment service.list and let that create the page filter and call the environmentFilterService etc.

File: 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java [hunk 8]
```diff
@@ -108,6 +111,69 @@ public Criteria createCriteriaForGetList(
     }
   }
 
+  public Criteria createCriteriaToGetScopedEnvironments(String accountId, List<String> orgIdentifiers,
```
Comment (puthrayaharness) [inline 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java:114]: Let's have a standard/common criteria or filter on top. We are duplicating this everywhere - 
```      
Criteria.where(EnvironmentKeys.accountId)
                        .is(accountId)
                        .and(EnvironmentKeys.projectIdentifier)
                        .is(null)
                        .and(EnvironmentKeys.deleted)
                        .is(false);
```

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

**Review by sarthakkasat — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54818)* status: 11 file(s) to review. 1 change requests. 12. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/environment/resources/EnvironmentResourceV2.java` line 521 at r3](https://reviewable.io/reviews/harness/harness-core/54818#gh-1379941577:-NiGSNnnD3o07CZ88msw:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/79ba9e9be69a195f5fccf9164212e0bc28ef85bc/120-ng-manager/src/main/java/io/harness/ng/core/environment/resources/EnvironmentResourceV2.java#L521)):*
<details><summary><i>Previously, puthrayaharness (Deepak Puthraya) wrote…</i></summary><blockquote>

I mean move the fetching criteria from environmentFilter service and then create the pageRequest and then call environmentService.

Instead, just call environment service.list and let that create the page filter and call the environmentFilterService etc.
</blockquote></details>

Done.
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java` line 65 at r3](https://reviewable.io/reviews/harness/harness-core/54818#gh-1379936219:-NiEtxPI7wIvVL9gdFTM:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/79ba9e9be69a195f5fccf9164212e0bc28ef85bc/127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java#L65)):*
<details><summary><i>Previously, puthrayaharness (Deepak Puthraya) wrote…</i></summary><blockquote>

We probably have the scopes already defined as constants, let's leverage that. No need to recreate these constants.
</blockquote></details>

Done.
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java` line 114 at r3](https://reviewable.io/reviews/harness/harness-core/54818#gh-1379944400:-NiGSOuL2nr0ZAEZzlyp:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/79ba9e9be69a195f5fccf9164212e0bc28ef85bc/127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java#L114)):*
<details><summary><i>Previously, puthrayaharness (Deepak Puthraya) wrote…</i></summary><blockquote>

Let's have a standard/common criteria or filter on top. We are duplicating this everywhere - 
```      
Criteria.where(EnvironmentKeys.accountId)
                        .is(accountId)
                        .and(EnvironmentKeys.projectIdentifier)
                        .is(null)
                        .and(EnvironmentKeys.deleted)
                        .is(false);
```
</blockquote></details>

Done.
___
*[`960-ng-core-beans/src/main/java/io/harness/ng/core/environment/dto/ScopedEnvironmentResponseDTO.java` line 2 at r2](https://reviewable.io/reviews/harness/harness-core/54818#gh-1379525308:-NiEuNUOEB6dtjSZYq-G:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/26b77f842382ba6772986c11d6b9008924ed73f3/960-ng-core-beans/src/main/java/io/harness/ng/core/environment/dto/ScopedEnvironmentResponseDTO.java#L2)):*
<details><summary><i>Previously, lovish1999 (Lovish Bansal) wrote…</i></summary><blockquote>

:nit: 2023
</blockquote></details>

Done.

File: 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java [hunk 9]
```diff
@@ -108,6 +111,59 @@ public Criteria createCriteriaForGetList(
...
+        projectCriteria.and(EnvironmentKeys.projectIdentifier).in(projectIdentifiers);
+      } else {
+        projectCriteria.and(EnvironmentKeys.projectIdentifier).exists(true);
+      }
+      orCriteriaList.add(projectCriteria);
+    }
+    return new Criteria().orOperator(orCriteriaList);
+  }
+
+  private static void setScopeCriteria(String accountId, String envType, Criteria accountCriteria) {
```
Comment (tchaurasiya) [inline 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java:160]: static not required

File: 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java [hunk 10]
```diff
@@ -108,6 +111,59 @@ public Criteria createCriteriaForGetList(
...
+      setScopeCriteria(accountId, envType, accountCriteria);
+      orCriteriaList.add(accountCriteria);
+    }
+    if (scopes.stream().anyMatch(scope -> scope.equalsIgnoreCase(ORG))) {
+      orgCriteria = Criteria.where(EnvironmentKeys.projectIdentifier).is(null);
+      setScopeCriteria(accountId, envType, orgCriteria);
+      if (isNotEmpty(orgIdentifiers)) {
+        orgCriteria.and(EnvironmentKeys.orgIdentifier).in(orgIdentifiers);
+      } else {
+        orgCriteria.and(EnvironmentKeys.orgIdentifier).exists(true);
```
Comment (tchaurasiya) [inline 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java:138]: Why are we adding exist true, insteas shouldn't we add not null
https://www.mongodb.com/community/forums/t/mongodb-bytes-6-querying-when-one-field-is-null-or-does-not-exist/204963

File: 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentMapper.java [hunk 11]
```diff
@@ -195,6 +196,21 @@ public EnvironmentResponse toResponseWrapper(Environment environment) {
         .lastModifiedAt(environment.getLastModifiedAt())
         .build();
   }
+  public ScopedEnvironmentResponseDTO toScopedResponseWrapper(Environment environment) {
+    return ScopedEnvironmentResponseDTO.builder()
+        .id(environment.getId())
```
Comment (tchaurasiya) [inline 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentMapper.java:201]: id,version are not needed

File: 960-ng-core-beans/src/main/java/io/harness/ng/core/environment/dto/ScopedEnvironmentResponseDTO.java [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+import lombok.experimental.FieldDefaults;
+
+@CodePulse(module = ProductModule.CDS, unitCoverageRequired = false,
+    components = {HarnessModuleComponent.CDS_SERVICE_ENVIRONMENT})
+@Data
+@Builder
+@FieldDefaults(level = AccessLevel.PRIVATE)
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class ScopedEnvironmentResponseDTO {
```
Comment (tchaurasiya) [inline 960-ng-core-beans/src/main/java/io/harness/ng/core/environment/dto/ScopedEnvironmentResponseDTO.java:30]: id, version are not required here

**Review by tchaurasiya — CHANGES_REQUESTED** [review]

**Review by sarthakkasat — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54818)* status: 11 file(s) to review. 2 change requests. 16. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/environment/resources/EnvironmentResourceV2.java` line 519 at r4](https://reviewable.io/reviews/harness/harness-core/54818#gh-1381251301:-NiJRrmt4L-VXP_hWcoh:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/dd0627827c08d7bddcc1debce18c19d9a0ec7ef9/120-ng-manager/src/main/java/io/harness/ng/core/environment/resources/EnvironmentResourceV2.java#L519)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

nit: This is the body for filter properties like list of orgIds, projectIds and Scopes
</blockquote></details>

Done
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java` line 138 at r4](https://reviewable.io/reviews/harness/harness-core/54818#gh-1381270475:-NiJT3Nn1SyUdwMwzJgb:bp05z38) ([raw file](https://github.com/harness/harness-core/blob/dd0627827c08d7bddcc1debce18c19d9a0ec7ef9/127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java#L138)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

Why are we adding exist true, insteas shouldn't we add not null
https://www.mongodb.com/community/forums/t/mongodb-bytes-6-querying-when-one-field-is-null-or-does-not-exist/204963
</blockquote></details>

Updated the field to use $ne 
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java` line 160 at r4](https://reviewable.io/reviews/harness/harness-core/54818#gh-1381264186:-NiJSDS_8TFtobwZyjAQ:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/dd0627827c08d7bddcc1debce18c19d9a0ec7ef9/127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelper.java#L160)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

static not required
</blockquote></details>

Done.
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentMapper.java` line 201 at r4](https://reviewable.io/reviews/harness/harness-core/54818#gh-1381272331:-NiJTGMT15ZUwvgTwpJ0:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/dd0627827c08d7bddcc1debce18c19d9a0ec7ef9/127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/mappers/EnvironmentMapper.java#L201)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

id,version are not needed
</blockquote></details>

Done.
___
*[`960-ng-core-beans/src/main/java/io/harness/ng/core/environment/dto/ScopedEnvironmentResponseDTO.java` line 30 at r4](https://reviewable.io/reviews/harness/harness-core/54818#gh-1381273184:-NiJTHWS3KLSJ2BanHe5:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/dd0627827c08d7bddcc1debce18c19d9a0ec7ef9/960-ng-core-beans/src/main/java/io/harness/ng/core/environment/dto/ScopedEnvironmentResponseDTO.java#L30)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

id, version are not required here
</blockquote></details>

Done.

**Review by tchaurasiya — DISMISSED** [review]

**Review by tchaurasiya — APPROVED** [review]

**Review by puthrayaharness — APPROVED** [review]

**Comment by sarthakkasat** [comment]
trigger uts, trigger review, trigger messagecheck

**Comment by sarthakkasat** [comment]
trigger uts

**Review by puthrayaharness — APPROVED** [review]

**Review by tchaurasiya — APPROVED** [review]

**Comment by puthrayaharness** [comment]
trigger review

**Comment by sarthakkasat** [comment]
trigger uts

**Comment by sarthakkasat** [comment]
trigger checkstyle

**Comment by sarthakkasat** [comment]
trigger ut6

**Pull request merged**